cmd/gossamer/config_test.go

```diff
@@ -417,341 +399,290 @@ func TestCoreConfigFromFlags(t *testing.T) {
+				DiscoveryInterval: defaultNetworkCfg.DiscoveryInterval,
+				MinPeers:          defaultNetworkCfg.MinPeers,
+				MaxPeers:          defaultNetworkCfg.MaxPeers,
 			},
 		},
-		{
-			"Test gossamer --pubip",
-			[]string{"config", "pubip"},
-			[]interface{}{testCfgFile, "10.0.5.2"},
+		"Test_gossamer_--no-mdns": {
+			[]string{"app", "--no-mdns"},
+			dot.NetworkConfig{
+				Port:              defaultNetworkCfg.Port,
+				NoMDNS:            true,
+				DiscoveryInterval: defaultNetworkCfg.DiscoveryInterval,
+				MinPeers:          defaultNetworkCfg.MinPeers,
+				MaxPeers:          defaultNetworkCfg.MaxPeers,
+			},
+		},
+		"Test_gossamer_--pubip": {
+			[]string{"app", "--pubip", "10.0.5.2"},
 			dot.NetworkConfig{
-				Port:              testCfg.Network.Port,
-				Bootnodes:         testCfg.Network.Bootnodes,
-				ProtocolID:        testCfg.Network.ProtocolID,
-				NoBootstrap:       testCfg.Network.NoBootstrap,
-				NoMDNS:            false,
-				DiscoveryInterval: time.Second * 10,
-				MinPeers:          testCfg.Network.MinPeers,
-				MaxPeers:          testCfg.Network.MaxPeers,
+				Port:              defaultNetworkCfg.Port,
+				DiscoveryInterval: defaultNetworkCfg.DiscoveryInterval,
+				MinPeers:          defaultNetworkCfg.MinPeers,
+				MaxPeers:          defaultNetworkCfg.MaxPeers,
 				PublicIP:          "10.0.5.2",
 			},
 		},
-		{
-			"Test gossamer --pubdns",
-			[]string{"config", "pubdns"},
-			[]interface{}{testCfgFile, "alice"},
+		"Test_gossamer_--pubdns": {
+			[]string{"app", "--pubdns", "alice"},
 			dot.NetworkConfig{
-				Port:              testCfg.Network.Port,
-				Bootnodes:         testCfg.Network.Bootnodes,
-				ProtocolID:        testCfg.Network.ProtocolID,
-				NoBootstrap:       testCfg.Network.NoBootstrap,
-				NoMDNS:            false,
-				DiscoveryInterval: time.Second * 10,
-				MinPeers:          testCfg.Network.MinPeers,
-				MaxPeers:          testCfg.Network.MaxPeers,
+				Port:              defaultNetworkCfg.Port,
+				DiscoveryInterval: defaultNetworkCfg.DiscoveryInterval,
+				MinPeers:          defaultNetworkCfg.MinPeers,
+				MaxPeers:          defaultNetworkCfg.MaxPeers,
 				PublicDNS:         "alice",
 			},
 		},
+		"Test_gossamer_--node-key": {
+			[]string{"app", "--node-key", "testkey"},
+			dot.NetworkConfig{
+				Port:              defaultNetworkCfg.Port,
+				DiscoveryInterval: defaultNetworkCfg.DiscoveryInterval,
+				MinPeers:          defaultNetworkCfg.MinPeers,
+				MaxPeers:          defaultNetworkCfg.MaxPeers,
+				NodeKey:           "testkey",
+			},
+		},
 	}
 
-	for _, c := range testcases {
+	for key, c := range testcases {
 		c := c // bypass scopelint false positive
-		t.Run(c.description, func(t *testing.T) {
-			ctx, err := newTestContext(c.description, c.flags, c.values)
-			require.NoError(t, err)
-			cfg, err := createDotConfig(ctx)
-			require.NoError(t, err)
-			require.Equal(t, c.expected, cfg.Network)
+		t.Run(key, func(t *testing.T) {
```

Comment:
nit run in parallel

```suggestion
		t.Run(key, func(t *testing.T) {
			t.Parallel()

```

you can then remove the comment `// bypass scopelint false positive`